enhancements/ingress/externaldns-operator.md

```diff
@@ -0,0 +1,261 @@
+* The operator will be capable of observing other resources to perform basic sequencing required for OpenShift
+integration.
+
+An example can make this easier to understand, here is what a CRD instance for ExternalDNS may look like:
+
+```yaml
+apiVersion: operator.externaldns.io/v1alpha1
+kind: externaldns
+metadata:
+  name: default
+  namespace: my-externaldns-namespace
+spec:
+  replicas: 2
+status:
+  availableReplicas: 2
+  conditions:
+  - lastTransitionTime: "2020-08-20T23:01:33Z"
+    reason: Valid
+    status: "True"
+    type: Admitted
+  - lastTransitionTime: "2020-08-20T23:07:05Z"
+    status: "True"
+    type: Available
+  - lastTransitionTime: "2020-08-20T23:07:05Z"
+    message: The deployment has Available status condition set to True
+    reason: DeploymentAvailable
+    status: "False"
+    type: DeploymentDegraded
+```
+
+This manifest creates a deployment that ensures 2 instances of ExternalDNS are always running in the cluster.
+Additional `spec` and/or `status` fields may be introduced based on requirements.
+
+* Introduce a CRD for managing DNS records that satisfies the ExternalDNS
+[CRD source](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/contributing/crd-source.md). __Note:__ The
+[ingressoperator/dnsrecord](https://github.com/openshift/api/blob/master/operatoringress/v1/types.go) CRD does not meet
+ExternalDNS CRD source requirements. If the ingressoperator/dnsrecord can be refactored, use this CRD instead of
+introducing a new CRD to manage DNS records.
+* Common fields in spec that define the desired configuration of an external DNS record.
+* Common fields in status that expose the actual state of the DNS record.
+* Platform components that require external DNS will use this CRD for managing DNS records.
+
+An example can make this easier to understand, here is what a CRD instance for an external DNS record may look like:
+
+```yaml
+apiVersion: externaldns.operator.openshift.io/v1alpha1
+kind: dnsrecord
+metadata:
+  name: default
+  namespace: my-component-namespace
+spec:
+  endpoints:
+  - dnsName: foo.example.com
+    targets:
+    - 1.2.3.4
+    recordType: A
+    recordTTL: 30
+status:
+  zones:
+  - conditions:
+    - lastTransitionTime: "2020-08-26T16:24:41Z"
+      message: The DNS provider succeeded in ensuring the record
+      reason: ProviderSuccess
+      status: "False"
+      type: Failed
+    dnsZone:
+      id: <MY_PROVIDER_ZONE_ID>
+```
+
+This manifest creates a DNS A record for hostname "foo.example.com" that resolves to address "1.2.3.4" in the configured
+provider. Additional `spec` and/or `status` fields may be introduced based on requirements.
+
+__Note:__ Any API(s) introduced by this enhancement will follow the Kubernetes API
+[deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/).
+
+### User Stories [optional]
+
+#### Story 1
+
+As a developer, I need the ability to manage DNS records for OpenShift routes.
```

Comment:
This story seems a little vague. How about instead:
`As a developer, I want to manage DNS records for OpenShift Routes via my OpenShift cluster, so that I do not have to spend time manually configuring my cloud provider's DNS.`